doc/decoding/going_further.rst

```diff
@@ -0,0 +1,290 @@
+.. for doctests to run, we need to define variables that are define in
```

Comment:
I think that this should be an example. Ideally reusing an existing one, rather than adding a new one. We have plenty already.